src/api.cpp

```diff
@@ -4965,10 +5005,23 @@ cl_int CLVK_API_CALL clEnqueueReadImage(
         return CL_INVALID_OPERATION;
     }
 
-    return cvk_enqueue_image_copy(command_queue, CL_COMMAND_READ_IMAGE, image,
-                                  blocking_read, origin, region, row_pitch,
-                                  slice_pitch, ptr, num_events_in_wait_list,
-                                  event_wait_list, event);
+    auto internal_image = static_cast<cvk_image*>(image);
+    if (internal_image->is_backed_by_buffer_view()) {
+        auto cmd = new cvk_command_buffer_host_copy(
+            command_queue, CL_COMMAND_READ_BUFFER,
```

Comment:
```suggestion
            command_queue, CL_COMMAND_READ_IMAGE,
```

Otherwise, the application could see the wrong command type.

Reply:
Done